### PR TITLE
Add support for getting authenticated federation media

### PR DESCRIPTION
**Pull request by S7evinK**

### CODE REVIEW

File: fclient/federationclient.go [hunk 1]
```diff
@@ -736,3 +738,31 @@ func (ac *federationClient) RoomHierarchy(
 	}
 	return
 }
+
```
Comment (HenrikSolver) [inline fclient/federationclient.go:741]: I think documentation in the same spirit as the other methods would be appropriate.

**Review by HenrikSolver — COMMENTED** [review]
To my understanding this looks fine.

File: fclient/federationclient.go (hunk 1 above)
Comment (S7evinK) [reply fclient/federationclient.go:741]: Fair, and good to document that the caller of this method is responsible to close the returned response body, if any.

**Review by HenrikSolver — APPROVED** [review]
LGTM

**Comment by S7evinK** [comment]
Thanks for the review! :)

**Pull request merged**